### PR TITLE
docs (notifications a11y): update to new template

### LINKED ISSUE #4253: [Callout] website docs: a11y tab
## Acceptance criteria

- The Notification component [Accessibility tab](https://carbondesignsystem.com/components/notification/accessibility/) needs to be updated with the new [content template](https://carbondesignsystem.com/contributing/documentation/#accessibility-template).
- Michael Gower will be a good reference when working on this issue.

## A11Y template
Look at other accessibility tabs to understand what type of content we are expecting on these pages.

### Page description
- [x] Determine which page description in the [content template](https://carbondesignsystem.com/contributing/documentation/#accessibility-template) is most appropriate for Notification. This can be determined towards the end of the process.  

### What Carbon provides
This section documents any existing accessibility features that come out of the box in the carbon code, like keyboard navigation, focus order, regions, ect. 
- [x] Consult with developers and Michael Gower to determine what existing accessibility features currently exist for all variants of Notification. 
- [x] Write an H3 section for each accessibility feature. 
- [x] Provide an image that annotates or depicts the accessibility interaction for each H3 section. 

### Design recommendation
This section is less straightforward and will require close collaboration with Michael Gower. In this section, you will give guidance on what users will need to consider when designing with Notification that may require additional design attention from product users to create an accessible experience. Not all components will have this section. 

- [x] Consult with Michael Gower to determine what accessibility design recommendations need to be communicated for Notification. 
- [x] Write an H3 section for each design recommendation. 
- [x] Provide an image that annotates or depicts the design recommendation for each H3 section. 

### Developement considerations 
Some of this information may already be available in the [existing a11y tab](https://carbondesignsystem.com/components/notification/accessibility/#how-it-works) content.

- [x] Consult with developers and Michael Gower to determine what the developer considerations 
- [x] Provide a bulleted list of implementation considerations for ensuring accessibility. If there is the ability to reference the IBM toolkit or the ARIA authoring practices, that should be done.

### PR DESCRIPTION
**Pull request by benjamin-kurien**
Closes #4253

Accessibility document for notification, updated to follow the new template.

#### Changelog

**New**

- Keyboard interactions for all notification types, with images.
- Design recommendations for semantic styling
- Development considerations

**Changed**

- Updated to new template

**Removed**

- Irrelevant sections from previous template was removed. Data was retained and moved to the relevant sections under the new template.

### CODE REVIEW

File: src/pages/components/notification/images/notification-accessibility-actionable.png
Comment (alina-jacob) [inline src/pages/components/notification/images/notification-accessibility-actionable.png:1]: Hi @benjamin-kurien, just adding a comment on the images. The tab stop curvature looked a little inconsistent (based on the perpendicular logic) so I have attempted an exploration [here](https://www.figma.com/design/NizIyBLaWztxQDnsBzOKbY/Callout?node-id=1112-12152&t=9KymuWPl7omWtLVl-4) (but, this is in 3:2 aspect ratio and not the 16:9 as recommended in the crit-call)

So totally your call! 
image
image

Comment (aagonzales) [reply src/pages/components/notification/images/notification-accessibility-actionable.png:1]: Oh, this was a good catch, @alina-jacob. You are totally right. We normally want the line to start going straight up for just a little bit before it starts curving. This is why we need to make some components for these arrows. It's easy to miss those nuances. I didn't even notice it at first.  
image

**Review by aagonzales — APPROVED** [review]
This all looks good to me except that one image tweak @alina-jacob found. Once that is fixed its good from my perspective.

**Comment by mbgower** [comment]
@benjamin-kurien as I mentioned on a prior call, the accessibility content is normally built to complement information already provided in the Usage and Style tabs; however, when I look at the deployment preview, there is no occurrence of the word "Callout" on either of these pages. The only place it is covered is the a11y tab.
Is that just a result of those changes being in a separate PR? If so, please provide, as it is hard to review this material without knowing what the user has already been told about Callouts on other tabs.

**Comment by benjamin-kurien** [comment]
> @benjamin-kurien as I mentioned on a prior call, the accessibility content is normally built to complement information already provided in the Usage and Style tabs; however, when I look at the deployment preview, there is no occurrence of the word "Callout" on either of these pages. The only place it is covered is the a11y tab. Is that just a result of those changes being in a separate PR? If so, please provide, as it is hard to review this material without knowing what the user has already been told about Callouts on other tabs.

Hi @mbgower Here's a link to the [PR ](https://github.com/carbon-design-system/carbon-website/pull/4218) that documents the usage and style guidance for Callout

**Review by tay1orjones — APPROVED** [review]

**Comment by mbgower** [comment]
Here's some comments. Recommend downloading and opening with Acrobat Reader to more easily review.
[Notification – Carbon Design System.pdf](https://github.com/user-attachments/files/17214942/Notification.Carbon.Design.System.pdf)

**Comment by mbgower** [comment]
My apologies for not just putting these comments inline in the PR!

**Comment by benjamin-kurien** [comment]
Hi @mbgower. I have updated the PR with the changes you requested. Please review

**Pull request merged**